cloud_functions/ads_policy_monitor/google_ads.py

```diff
@@ -95,14 +108,65 @@ def run_gaarf_report(payload: models.Payload,
                                  report_fetcher=report_fetcher,
                                  customer_ids=payload.customer_ids)
 
-    path = f'gaql/{report_config.gaql_filename}'
+    if report_config.is_asset_report:
+        all_reports = []
+        for gaql_filename in report_config.gaql_filenames:
+            path = f'gaql/{gaql_filename}'
+            all_reports.append(
+                run_query_from_file(
+                    query_path=path,
+                    report_fetcher=report_fetcher,
+                    customer_ids=payload.customer_ids,
+                ))
+        return combine_assets_reports(all_reports)
+
+    path = f'gaql/{report_config.gaql_filenames}'
     return run_query_from_file(
         query_path=path,
         report_fetcher=report_fetcher,
         customer_ids=payload.customer_ids,
     )
 
 
+def combine_assets_reports(gaarf_reports: List[GaarfReport]) -> GaarfReport:
+    """Combine assets reports.
+
+    Args:
+        gaarf_reports: a list of gaarf reports.
+
+    Returns:
+        A GAARF report.
+    """
+    logger.info('Combining Assets reports')
+    combined_assets_report = pd.DataFrame()
+    for report in gaarf_reports:
+        report_df = report.to_pandas()
+
+        if report_df.empty:
+            continue
+
+        if ("ad_group_id" in report_df.keys() and
+                "campaign_id" in report_df.keys()):
+            report_df["asset_level"] = "Ad Group"
+        elif ("campaign_id" in report_df.keys()):
+            report_df["asset_level"] = "Campaign"
+        else:
+            report_df["asset_level"] = "Account"
+
+        # Convert policy_topic_entries to str (otherwise group by will fail)
+        report_df[
```

Comment:
Are you not able to use the `astype` cast here?

```
df['asset_policy_summary_policy_topic_entries_topics'] = df['asset_policy_summary_policy_topic_entries_topics'].astype(str)
```

Reply:
It either returns a string or a list of strings. If I do astype it won't add delimiter that would be consistent with Gaarf (in Gaarf it separates list values with "|")

Reply:
Ah ok. Makes sense. Thanks for confirming.